### DISCUSSION TITLE
Bulk import

### DISCUSSION THREAD

**Discussion by edkedk99** [immich-app/immich#3929] [Q&A]
I have used immich cli tool to import my existing photos to immich. It seems to work fine, it doesn't give me any error but there is much less photos in the immich library photos than in my source directory whre I uploaded from.

I am afraid immich cli tool might have skipped some file and I am afraid immich doesn't have my full photo collection. I suspect it might be because my source directory might have some duplicates.

Can I be confident that all photos and videos was uploaded if there was no error? If not, is the any way to check what could be the missing files?

My library folder (./library/library/admin) has 24000 files while my source directory has 39879.

What is interesting is that immich cli tool correctly detects 39876 image files in source directory, but for some reason the library end up with only 24000 even as I ran the tool multiple times.

I appreciate any help on it.

**Comment by ScribbleGhost** [comment]
- Are you sure they are not duplicates?
- Are you sure they are supported file types?
- Are you sure they actually were not imported?

Unfortunately it is impossible to debug this without any more information so I don't see this post going anywhere unless someone else have similar experiences.
